### ISSUE TITLE
Remove Favorite on Soft-Delete elements

### ISSUE THREAD

**Issue by Bonapara** [twentyhq/core-team-issues#107]
# Current behavior

I can add a soft deleted item as favorite. This result in creating an "empty favorite"

![CleanShot 2024-10-14 at 11 13 29](https://github.com/user-attachments/assets/c1ba7c61-3b29-4166-9482-bc20df8cfc23)

# Desired Behavior

1.  Hide soft-deleted items from left sidebar.
2. Remove the ability to add a soft-deleted item as favorite

![13c54269c187e176138ce5fd7dc026f3123dd9c02967a1958608e4131de96e32](https://github.com/user-attachments/assets/d9590bb4-48ca-4eb9-b682-1f05f76579b2)

![CleanShot 2024-10-14 at 11 14 38](https://github.com/user-attachments/assets/24cf41d3-0d1f-4ef1-a8aa-e15771667f2a)

**Comment by NitinPSingh** [comment]
is it for core team?

**Comment by Karankhatik** [comment]
@Bonapara  if this is not meant for the core team, could you please assign it to me?
@NitinPSingh , if you'd like to work on this, I will take up another issue.

**Comment by Weiko** [comment]
@Bonapara I think we can safely delete the favorite (in the DB) if a record is soft-delete. Wdyt?

Note: What about favorites linked to objects? I guess it's the same (even more true since objects can't be soft-deleted, except if we want to treat them as soft-deleted if they are deactivated then I have the same question)

**Comment by Bonapara** [comment]
Yes, we should, but it seems to be a significant feature to implement correctly, as it would involve cascading...

So I think V1 is "hide" and V2 is "cascading"

**Comment by charlesBochet** [comment]
This should be properly done through Cascade Soft Delete and Cascade Destory at ORM layer
for now I think this is already handled "locally" for favorites. Should be done, to be checked

**Comment by prastoin** [comment]
## Update
Fixed in production, we cannot add deleted record as favorite anymore:

![Image](https://github.com/user-attachments/assets/99440f16-61f2-4008-a04f-11734e09fd51)

## Issue
Encountered a bug in production that seems to be fixed on main, but still needs to be addressed in next release

Repro:
- Delete favorite record
- Filter view by deleted records
- Open cmd+K panel

Open question: Is it an expected behavior for the just deleted record row to still be selected after filtering by deleted records ?

https://github.com/user-attachments/assets/1cd8e7f2-4799-439f-8e79-8682c9e29974

**Comment by prastoin** [comment]
Closing this one from now, @charlesBochet what would you recommend for the above what seems to be production-only bug ? Please let me know
